src/page-weekly-snapshot/LocaleSummaryTable.tsx

```diff
@@ -0,0 +1,368 @@
+  width: 100%;
+  margin: 15px 0 15px;
+  display: flex;
+  justify-content: space-between;
+  align-items: baseline;
+  color: ${Colors.black};
+`;
+
+const Right = styled.div`
+  text-align: right;
+`;
+
+const Left = styled.div`
+  text-align: left;
+  font-size: 24px;
+  font-family: "Libre Baskerville";
+`;
+
+const BorderDiv = styled.div`
+  border-top: 2px solid ${Colors.black};
+`;
+
+const TableCell = styled.td<{ label?: boolean }>`
+  font-size: 11px;
+  line-height: 200%;
+  text-align: "left";
+  width: ${(props) => (props.label ? "200px" : "auto")};
+`;
+
+type StateMetrics = {
+  stateName: string;
+  casesPerCapita: number | undefined;
+  deathsPerCapita: number | undefined;
+};
+
+type incarceratedData = {
+  incarceratedData: number;
+  hasData: boolean;
+};
+
+const POPULATION_SIZE = 100000;
+const NUM_STATES = 50;
+
+const VALUE_MAPPING = {
+  deaths: deathBracketKeys,
+  cases: caseBracketKeys,
+  population: incarceratedPopulationKeys,
+};
+
+function formatOrdinal(value: number) {
+  let suffix = ["th", "st", "nd", "rd"],
+    rem = value % 100;
+  return value + (suffix[(rem - 20) % 10] || suffix[rem] || suffix[0]);
+}
+
+function getPerCapita(
+  numerator: number | undefined,
+  denominator: number | undefined,
+) {
+  let result = 0;
+  if (numerator && denominator) {
+    result = Math.round((numerator / denominator) * POPULATION_SIZE);
+  }
+  return result;
+}
+
+function getAllStateData(localeData: LocaleData, stateNames: string[]) {
+  const stateMetrics: StateMetrics[] = [];
+  // D.C. isn't a state!
+  stateNames = stateNames.filter((item) => item !== "District of Columbia");
+
+  for (let i = 0; i < stateNames.length; i++) {
+    const localeDataStateTotal = localeData?.get(stateNames[i])?.get("Total");
+    if (localeDataStateTotal) {
+      const totalCases = localeDataStateTotal["reportedCases"];
+      const totalPopulation = localeDataStateTotal["totalPopulation"];
+      const totalDeaths = localeDataStateTotal["totalDeaths"];
+      const casesPerCapita = getPerCapita(totalCases, totalPopulation);
+      const deathsPerCapita = getPerCapita(totalDeaths, totalPopulation);
+      stateMetrics.push({
+        stateName: stateNames[i],
+        casesPerCapita: casesPerCapita,
+        deathsPerCapita: deathsPerCapita,
+      });
+    }
+  }
+  return stateMetrics;
+}
+
+function getStateRank(
+  stateTotals: StateMetrics[],
+  selectedStateName: string | undefined,
+  filterValue: string,
+) {
+  const rankedStates = orderBy(
+    stateTotals.filter((c) => !!get(c, filterValue)),
+    [filterValue],
+    ["asc"],
+  );
+
+  for (let i = 0; i < rankedStates.length; i++) {
+    const currState = rankedStates[i];
+    const currStateDataPerCapita = get(currState, filterValue);
+    if (
+      selectedStateName &&
+      currState.stateName === selectedStateName &&
+      currStateDataPerCapita
+    ) {
+      return [currStateDataPerCapita, i + 1];
+    }
+  }
+  return [-1, -1];
```

Comment:
this return value is pretty obscure ... it is not immediately obvious why it takes two numbers to represent the state's rank. could this be an object instead with more explicit keys?